### PR TITLE
feat: pfe-tabs keep history on tab click and open tab based on url parameters

### LINKED ISSUE #562: pfe-tabs - URL access to a specific tab
For course training pages, we would like the ability to have a deep link to a specific tab on the page.

### PR DESCRIPTION
**Pull request by kylebuch8**
Test by opening this demo page: http://localhost:8000/elements/pfe-tabs/demo/#tab-history

Click on each of the tabs and you should see the URL update with the id of the selected tab.

To test the feature where the correct tab opens on page load based on a query string parameter, open this page: http://localhost:8000/elements/pfe-tabs/demo/?my-tabs=tab3#tab-history. The third tab should be open on page load.

Fixes #562

### CODE REVIEW

**Comment by kylebuch8** [comment]
For anyone who wants to see this on Netlify:
https://deploy-preview-786--happy-galileo-ea79c4.netlify.com/elements/pfe-tabs/demo/?my-tabs=tab2#tab-history

File: elements/pfe-tabs/README.md [hunk 1]
```diff
@@ -68,6 +68,62 @@ Changes the context of the call-to-action to one of 3 possible themes:
...
+  <pfe-tab-panel role="region" slot="panel">
+    <h2>Content 2</h2>
+    <p>Tab 2 panel content.</p>
+  </pfe-tab-panel>
+</pfe-tabs>
+```
+
+*Note:* This feature is not supported in IE11.
+
+## Using the URL to open a specific tab
```
Comment (castastrophe) [inline elements/pfe-tabs/README.md]: Nice documentation addition!

Comment (kylebuch8) [reply elements/pfe-tabs/README.md]: Thanks!

File: elements/pfe-tabs/src/pfe-tabs.js [hunk 2]
```diff
@@ -373,6 +431,33 @@ class PfeTabs extends PFElement {
...
     this.selectedIndex = this._getTabIndex(event.currentTarget);
   }
+
+  _getTabIndexFromURL() {
+    let urlParams;
+    let tabIndex = -1;
+
+    // @IE11 doesn't support URLSearchParams
+    // https://caniuse.com/#search=urlsearchparams
+    if (window.URLSearchParams) {
```
[the remark this answers is not in the document]
Comment (kylebuch8) [reply elements/pfe-tabs/src/pfe-tabs.js]: I moved this to a constant outside the scope of the class.

File: elements/pfe-tabs/src/pfe-tabs.js [hunk 3]
```diff
@@ -169,6 +197,23 @@ class PfeTabs extends PFElement {
...
+      this.tabHistory &&
+      this._updateHistory &&
+      this._canUseURLSearchParams
+    ) {
+      // rebuild the url
+      const pathname = window.location.pathname;
+      const urlParams = new URLSearchParams(window.location.search);
+      const hash = window.location.hash;
+
+      urlParams.set(this.id, tab.id);
```
[the remark this answers is not in the document]
Comment (kylebuch8) [reply elements/pfe-tabs/src/pfe-tabs.js]: Done.

File: elements/pfe-tabs/src/pfe-tabs.js [hunk 4]
```diff
@@ -373,6 +431,33 @@ class PfeTabs extends PFElement {
...
+
+    // @IE11 doesn't support URLSearchParams
+    // https://caniuse.com/#search=urlsearchparams
+    if (window.URLSearchParams) {
+      urlParams = new URLSearchParams(window.location.search);
+    }
+
+    if (urlParams && urlParams.has(this.id)) {
+      tabIndex = this._allTabs().findIndex(
+        tab => tab.id === urlParams.get(this.id)
```
[the remark this answers is not in the document]
Comment (kylebuch8) [reply elements/pfe-tabs/src/pfe-tabs.js]: Done.

**Pull request merged**